dvc/scm/git/tree.py

```diff
@@ -58,20 +74,29 @@ def open(self, path, mode="r", encoding="utf-8"):
         return io.StringIO(data.decode(encoding))
 
     def exists(self, path):
-        return self._git_object_by_path(path) is not None
+        if self._git_object_by_path(path) is None:
+            return False
+
+        return not self.dvcignore.is_ignored_file(
+            path
+        ) and not self.dvcignore.is_ignored_dir(path)
```

Comment:
just mimicing old CleanTree. Could reconsider whether or not we really need to deny direct access here. E.g. you could force `git add` for a gitignored file.